### ISSUE TITLE
Clarify "Role" field in "Verbatim Agent" attribute

### ISSUE THREAD

**Issue by genevieve-anderegg** [ArctosDB/arctos#4934]
Issue Documentation is http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html

**Is your feature request related to a problem? Please describe.**
Related to: https://github.com/ArctosDB/arctos/issues/4871
Verbatim collector has recently been changed to verbatim agent and moved up to the agent box on catalog record view. A new field called "Role" has been added, which is just the "Determination Method" field. However, this is unclear in both the data entry screen as well as when editing the record. This change from "verbatim collector" to "verbatim agent" also loses the information that this verbatim agent was specifically acting as the collector of the specimen (and not another role, such as preparator, donor, etc.). Would it be possible to clarify these fields as well as make a note that this information was moved from the previous "verbatim collector" attribute to the "verbatim agent" attribute?

Example record: https://arctos.database.museum/guid/DMNS:Inv:23805
![Capture1](https://user-images.githubusercontent.com/83782023/184397577-6e0d3210-f2fa-4319-8249-340390f04b4c.PNG)
![Capture](https://user-images.githubusercontent.com/83782023/184397645-7bb68ec0-d1e8-4a7c-b962-375aae1bbb4e.PNG)

This is also not clear in the data entry form, as any information you put in Method would go into Role.
![2](https://user-images.githubusercontent.com/83782023/184398622-cc5b267b-319e-4c72-aab0-60808e55f734.PNG)

**Describe what you're trying to accomplish** 
Maintain the verbatim collector information in the new Role field, and clarify that the Determination Method field is what is mapping into Role, and that is where the "verbatim collector" information should be stored.

**Describe the solution you'd like**
Maintain the verbatim collector information in the new Role field, and clarify that the Determination Method field is what is mapping into Role, and that is where the "verbatim collector" information should be stored.

**Describe alternatives you've considered**
I know this has probably discussed by the committee, apologies if I am missing any context and for jumping in on this after #4871 has already been discussed and implemented.

**Comment by genevieve-anderegg** [comment]
@sharpphyl @acdoll

**Comment by dustymc** [comment]
see

 https://github.com/ArctosDB/arctos/issues/4871  
https://github.com/ArctosDB/arctos/issues/4870 
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#verbatim_agent

**Comment by sharpphyl** [comment]
From [#4871](https://github.com/ArctosDB/arctos/issues/4871)
> Method is required, and should be used to specify (or approximate) in which role the agent is acting.
> IMPORTANT: This attribute would be auto-created as necessary by the agent merger scripts, into which the definition would be hard-coded. Help getting this right greatly appreciated.

Does this mean that the "Role" field should have been auto-created as "verbatim collector" rather than "unknown"?  Usually when you make a change of this type, @dustymc makes the changes and documents it.   Can you undo what you did and redo it with the Method field showing "verbatim collector"?

If we're expected to correct them, we need help.  I can't find a way to search for "verbatim agents" that now have the "unknown" role.  

<img width="861" alt="Screen Shot 2022-08-12 at 10 59 37 AM" src="https://user-images.githubusercontent.com/15368365/184407646-75a12d8b-d4d6-4125-8bfd-f63108be14f7.png">

If we can find them, and if we create a csv from the bulkload attribute template, will that overwrite "unknown" to "verbatim collector" or does it just add another attribute and we have to also delete the current attribute?  (The Manage/attributes function only adds an attribute.)

We will change our data entry instructions to select "verbatim agent" and put "verbatim collector" in the Method field so it will show up in the "Role" field.

**Comment by sharpphyl** [comment]
From [#4871](https://github.com/ArctosDB/arctos/issues/4871)

> **Migration Path**
> 
> For existing verbatim collector attributes with NULL method, update method to 'unspecified collector role' before migrating

Did this happen?  All our verbatim collectors now say "unknown" (rather than "verbatim collector") in the Role field unless there was data in the Method field.

<img width="688" alt="Screen Shot 2022-08-16 at 6 58 44 AM" src="https://user-images.githubusercontent.com/15368365/184885578-c3dabe7f-7956-4362-8651-ae6ae68cc929.png">

With data in the Method field, the Role becomes pretty weird.

<img width="676" alt="Screen Shot 2022-08-16 at 6 57 52 AM" src="https://user-images.githubusercontent.com/15368365/184885537-beb5cff5-e432-4fbb-b6bf-fddda6de1b12.png">

<img width="682" alt="Screen Shot 2022-08-16 at 6 58 04 AM" src="https://user-images.githubusercontent.com/15368365/184885552-94bbf05d-c8fd-496d-b0f7-4e65e98ef962.png">


Also, some of our records still show "verbatim collector" in the search results screen.

<img width="238" alt="Screen Shot 2022-08-16 at 7 02 08 AM" src="https://user-images.githubusercontent.com/15368365/184886578-017e491a-9d97-4b33-a981-80635e92adac.png">


But when you open the record https://arctos.database.museum/guid/DMNS:Inv:12602, it isn't there.


<img width="730" alt="Screen Shot 2022-08-16 at 7 02 28 AM" src="https://user-images.githubusercontent.com/15368365/184886314-9d5719b5-f0fe-4907-911f-92b344977aac.png">

Is there anyway to reconsider and add a field for "Role"?  We have had data in both the Method and Remark fields.  We have hours of work ahead of us to clean up this data.

**Comment by dustymc** [comment]
>'unspecified collector role'

Thanks! Done, that seems to have got hung up (in a sea of unicode conversion fail character errors...).

> still show "verbatim collector" in the search results screen.

They should clean up in a few days, or I can force the cache to refresh (but that'll shut down everything else until it happens).

> add a field for "Role"

This attribute acting on the record according to method is the "role."

There's no cleanup required, although having more standardized data will be useful if elevating these to Agents ever becomes a consideration. I'm happy to help, now or then. (FWIW I'm still not much of a fan of the ``collector`` [vocabulary](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollector_role), I'd probably just enter whatever it says on the label-or-whatever instead of trying to translate, but that's up to your policies.)

Let me know if I'm still not understanding something.

**Comment by Jegelewicz** [comment]
I think I brought this up before - but can we not just link units for this attribute to the collector vocabulary? This would provide a standard that everyone can use (they would have been using it on the low quality agent, so it seems like everyone could choose one of these?

**Comment by Jegelewicz** [comment]
Like @sharpphyl I have generally used "methods" here to say how the value was determined - from a tag from previously digitized data, etc. so mucking it up with the "role" seems like overload.

**Comment by dustymc** [comment]
> link units

[No.](https://handbook.arctosdb.org/documentation/attributes.html#attribute-units)

That's not structurally viable, and does not need stuffed into that fine of a pigeonhole - often all you'll know is "scribbled on label" and that's fine for these less-formal data. (That is, it's not just the names themselves that can benefit from some "verbatimizing.")

>seems like overload.

That's why we have definitions....

**Comment by sharpphyl** [comment]
> That's why we have definitions....

From https://handbook.arctosdb.org/documentation/attributes.html#attribute-units
It says here that **Method** is how the Attribute was determined. It does not say that Method is the Role of the verbatim agent.  

> This attribute acting on the record according to method is the "role."

In English, please!

> if elevating these to Agents ever becomes a consideration

No intention of elevating these to Agent.  That's why they are "verbatim agents."

> Let me know if I'm still not understanding something.

We still don't understand why the Role "verbatim collector"  was collapsed into the Method field.  We are losing the data we have Method in order to create a Role for our "verbatim agent."

@Jegelewicz 

> can we not just link units for this attribute to the collector vocabulary

Yes, if we can add "verbatim collector" to the collector vocabulary??

Please note that [#4871](https://github.com/ArctosDB/arctos/issues/4871) is titled **Code Table Request - verbatim agent** which is fine, but the implementation is creating problems.

**Comment by genevieve-anderegg** [comment]
> Like @sharpphyl I have generally used "methods" here to say how the value was determined - from a tag from previously digitized data, etc. so mucking it up with the "role" seems like overload.

That's exactly where I'm at as well. I don't think co-opting the method field for Role is the right way to go about this, because we often want to maintain both of those sets of information. The Role shows what an entity's relationship to a specimen is (verbatim collector, donor, etc.). Method is how we know what that relationship is (from the collectors catalog, original label, etc.). They are not the same piece of information, and acting as if they are is causing us to lose clarity in our data.

**Comment by genevieve-anderegg** [comment]
We all agree that getting rid of unnecessary low data agents is a great step forward, and using Verbatim agents is a great way to do that. However, we still want to maintain as much information about these verbatim agents as possible when we have it, and give them context in the catalog record. Having both Role and Method fields for Verbatim agents helps us keep track of the maximum amount of information we have about them. The current implementation of having "Method" (which is described in the handbook as as "how the attribute is determined") map into Role (what the verbatim agent's relationship to the specimen is) causes us to lose information, and forces us to put data that could be atomic into Remarks, which I think is something we all like to avoid.

**Comment by sharpphyl** [comment]
> There is some sort of deep misunderstanding here.

Yes, we're not following your reasoning, but I'm not sure what the priority of this issue is.  If we move a significant number of "low data agents" to the attribute "verbatim agent" this may become a bigger issue for more collections.

**Comment by Jegelewicz** [comment]
> > link units
> 
> [No.](https://handbook.arctosdb.org/documentation/attributes.html#attribute-units)
> 
> That's not structurally viable, and does not need stuffed into that fine of a pigeonhole - often all you'll know is "scribbled on label" and that's fine for these less-formal data. (That is, it's not just the names themselves that can benefit from some "verbatimizing.")

"Scribbled on label" is the method, this needs separating from what we are translating as what the "scribbled on label" name DID.

This leads me to believe that "verbatim agent" is too generic, just as "verbatim collector" was misleading. We need a designated spot to put the known or suspected role of the verbatim agent if we ever hope to actually clean up and make better agents OR we need at a minimum the following attributes in addition to verbatim agent (use only when role is unknown):

verbatim collector
verbatim preparator
verbatim creator
verbatim copyright holder
verbatim subject

**Comment by dustymc** [comment]
>not following your reasoning,

Ditto! From here, it looks like the structure of a few things ([attributes](https://arctos.database.museum/tblbrowse.cfm?tbl=attributes) and [collectors](https://arctos.database.museum/tblbrowse.cfm?tbl=collector), at least) is being mixed up and the old overview being given precedence over the recent documentation for no reason that I can make sense of. Following the documentation will result in consistent lossless data in the only scope that can possibly matter here, this particular attribute. I can't see anything that might lead to less-functional data, any other approach that's more defensible within the confines of our data structure, nor any reason to modify that structure. If I'm wrong about that or not considering something functional then this should probably be prioritized, if not then ???

**Comment by sharpphyl** [comment]
> From here, it looks like the structure of a few things ([attributes](https://arctos.database.museum/tblbrowse.cfm?tbl=attributes) and [collectors](https://arctos.database.museum/tblbrowse.cfm?tbl=collector), at least) is being mixed up and the old overview being given precedence over the recent documentation for no reason that I can make sense of.

I looked at your links on attributes and collectors and feel the complexity of all the various agent and verbatim agents options is way beyond our collection's use of "verbatim collector."  We just feel like we're losing some basic information about who picked up the shell and how we know that.  

> verbatim collector
> verbatim preparator
> verbatim creator
> verbatim copyright holder
> verbatim subject

If it can be done, these are would be great Roles for verbatim agents.  We just don't want to lose the Method field.

**Comment by Jegelewicz** [comment]
@dustymc now we have this

![image](https://user-images.githubusercontent.com/5725767/185259880-0e9265a6-bc0c-4b85-b918-421aed94a160.png)

So the "method" was in remarks - which is where it probably was for every verbatim collector and now we have "unspecified" when clearly it is not! The UWBM ones should get fixed as I work on their stuff, but I imagine this is true for nearly all the verbatim things.

Perhaps the role really belongs in remark?

**Comment by Jegelewicz** [comment]
FWIW I have taken to doing this in method

collector as transcribed from previous data

or

preparator as written on tag

**Comment by dustymc** [comment]
>now we have this

It's a free-text attribute, you can do just about whatever you want. This was created to track agent mergers, and those look like this:

<img width="853" alt="Screen Shot 2022-08-17 at 4 41 30 PM" src="https://user-images.githubusercontent.com/5720791/185262372-4ba5e6ea-88b8-4639-ace6-13e60c2bd409.png">

>collector as transcribed from previous data

Not exactly how I'd do it - the method is collecting, 'transcribed...' is - well, remarkable - but certainly nothing I'll object to.

**Comment by Jegelewicz** [comment]
>  the method is collecting, 'transcribed...' is - well, remarkable

This makes absolutely no sense whatever to me. The method is "transcribed" and I think that is how @sharpphyl sees it as well? Method is "how I came to this value" EVERYWHERE else in Arctos attributes. Making this one cryptically different is never going to function as we want it to.

**Comment by sharpphyl** [comment]
It still makes no sense that a field called **Det. Method** maps into the field **Role**, but if you can't have a different set of fields for verbatim agent, can you at least name the field **Det. Method/Role**?  

Can you also do the following:

1.  Concatenate the current data in the Det Method field with the data in the Remarks field.

2.  Put Verbatim collector into that Role field for all of our records that have a verbatim agent?  That used to be the name of this attribute.  Otherwise, we have to manually edit >1500 catalog records.

**Comment by dustymc** [comment]
>no sense

That's why we have documentation!

Relabeled "Method" in next release.

>expect to look with the change 

??????

**Comment by sharpphyl** [comment]
@dustymc  I'm working on verbatim agent records that we have to clean-up because they don't show the role of the verbatim agent (formerly "verbatim collector") and the remarks and method have to be combined, etc.  I'm trying to find just the records that have "unspecified collector role" as the Method/Role.  The search screen only returns a list of every record with a verbatim agent - regardless of whether we've clarified the role.

<img width="570" alt="Screen Shot 2022-08-22 at 3 16 48 PM" src="https://user-images.githubusercontent.com/15368365/186020729-9e9e223d-5665-4e98-a339-077bfbefbde9.png">

Creating the csv files to delete and reload takes forever because all the attributes download and have to be parsed out.

<img width="1437" alt="Screen Shot 2022-08-22 at 3 22 41 PM" src="https://user-images.githubusercontent.com/15368365/186021525-f4fe851a-fc64-4b45-84c8-da4704183bce.png">

Can you add the search field Role (aka Method)?
Is there anyway to download just the attributes I want?  Currently it downloads all attributes. 
Or can you write SQL that will bring up all our records with the Role "unspecified collector role"?
Or share some faster way to do this that I'm not aware of?

**Comment by dustymc** [comment]
FYI I still don't understand your concerns, so let me know if I can help more than I am.

Here's the data:

```
select
guid,
attribute_type as attribute,
attribute_value,
determined_date as attribute_date,
determination_method as attribute_meth,
getPreferredAgentName(determined_by_agent_id) as determiner,
remarks
from flat
inner join attributes on flat.collection_object_id=attributes.collection_object_id and attribute_type='verbatim agent'
where guid_prefix='DMNS:Inv' and
determination_method='unspecified collector role'
```

[temp_dmns_inv_atrs.csv.zip](https://github.com/ArctosDB/arctos/files/9403186/temp_dmns_inv_atrs.csv.zip)

>But the record shows

I set DMNS:Inv to refresh, the offer to force everything else still stands but it'll still take a few days of priority CPU.

> search field Role (aka Method) to the catalog search screen?

Sure, I'll try to get it in next release - please open a new issue if I get distracted.

**Comment by acdoll** [comment]
Thanks for this data (it's pretty much what we need, except the remarks provided are the Specimen/Object Remarks, not the attribute_remark; I was able to pull that separately).

 I think we've got a handle on this now. Our main concern was that we may have lost information stored in the Det. Method field when you added 'unspecific collector role', which we did not (that was only added to NULL values for this field). We will unload and reload all these verbatim agents to get the display to look how we want it (mainly that Role [aka. Det. Method] will say 'verbatim collector' and the attribute remarks will  contain any additional information).

**Comment by dustymc** [comment]
>that was only added to NULL

Correct, and thanks for the explanation!

**Comment by acdoll** [comment]
> Relabeled "Method" in next release.

Sorry, but we do prefer this to display Role (not Method) on the object page so that it corresponds more to the way the real Agents display
![image](https://user-images.githubusercontent.com/12507297/186492731-9ae98621-de71-4a24-a0b1-cc40c4b86655.png)

I believe Phyllis was suggesting changing the column label to "Det. Method/Role" in attribute Edit/entry table here:
![image](https://user-images.githubusercontent.com/12507297/186492989-735cbed4-3e90-4314-baf5-c9feed1d7733.png)
However, that might make things more confusing for all of the other attributes.

Would it be possible to display the 'verbatim agent' attribute edit fields on the Agent edit page? Or would that require a major re-development? Something like this would be ideal:
![image](https://user-images.githubusercontent.com/12507297/186494095-9b4ce925-bb87-423c-95a8-d60e1218fbd2.png)

**Comment by dustymc** [comment]
>display Role (not Method)

I think this needs more discussion. I'm prone to handwave labeling away as "just UI" but I think maybe users on average aren't quite that flexible, and I am slowly coming around - or trying to - to the idea that this should generally be fairly close to the underlying structure. It's easy to change, there's no technical anything involved here, I (obviously!) don't have solutions, but I'm not sure we can do anything at all without confusing someone.

>Edit/entry 

Yea, I think that is definitely too much.

>attribute edit fields on the Agent edit page? 

Hu, neat.I don't think that's technically out of reach. Are you thinking "also" or "instead"? (I think maybe I'd prefer 'also' but both seem to have plenty of potential for confusion - how to best avoid that?)

**Comment by sharpphyl** [comment]
Until two days ago, this field in the catalog record view was titled **Role** not **Method**.  It's not just confusing to our data entry volunteers but the public record (see below) is equally confusing.  The agent Webb's role is as the verbatim collector.  

I see the comment above the data "Agents acting only in method-specific role."  Should the field be named "Method-specified role"?  If so, what does that mean?

<img width="781" alt="Verbatim collector - method or role?" src="https://user-images.githubusercontent.com/15368365/186666829-e41a8347-cb9e-42af-b71e-04d7915f5184.png">

**Comment by acdoll** [comment]
> I'd prefer 'also'

I think 'also' would work well. That gives you the option to put it in if you're working on a bunch of other attributes, but if you're just looking to modify an existing you are probably more likely to hit the Edit link in the Agent box.

**Comment by dustymc** [comment]
@sharpphyl https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#verbatim_agent is the only authoritative doc - am I still being too free with the ' handwave labeling away as "just UI" ' thing?

Going active for verbatim agent in specdetail attributes tab.

**Comment by sharpphyl** [comment]
@dustymc  I appreciate that you're following the documentation.  Perhaps @lin-fred and the [Agents Committee](https://github.com/ArctosDB/arctos/projects/11#card-85249653) can review the documentation - specifically "Indicate the agent role with regard to the catalog object in the attribute method."   Perhaps this is because it's difficult to have different fields for each attribute.

For Verbatim Agents, we can train our volunteers to insert the "agent role" in the "attribute method" field.  But it's hard to explain to the public that the **Method** field is really the agent **Role** if the attribute is Verbatim agent.  I find that sentence confusing to even write.
